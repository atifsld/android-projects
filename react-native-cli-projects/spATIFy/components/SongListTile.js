import React from "react";
import { View, Image, Text, Pressable } from "react-native";
import { StyleSheet } from "react-native";

function SongListTile(song) {
    return (
        <Pressable android_ripple={{color: '#ccc'}}>
            <View style={styles.songCardView}>
                <View style={styles.songImageView}>
                    <Image style = {styles.albumArt}
                        source={{uri: song.song.songAlbumArtUrl}}/>
                </View>
                <View style={styles.songTextView}>
                    <Text style={styles.songNameView}>{song.song.songName}</Text>
                    <Text style={styles.songArtistsView}>{song.song.songArtists}</Text>
                    <Text style={styles.songDurationView}>{song.song.songDuration}</Text>
                </View>
            </View>
        </Pressable>
    )
}

const styles = StyleSheet.create({
    songCardView: {
        flex: 1,
        flexDirection: 'row',
        margin: 8,
        height: 100,
        backgroundColor: "#03DAC6",
        elevation: 8,
        borderRadius: 8,
        shadowColor: 'black',
        shadowOpacity: 0.25,
        shadowOffset: {width: 0, height: 2},
        shadowRadius: 8
    },
    albumArt: {
        height: 90,
        width: 90,
        borderRadius: 6
    },
    songImageView: {
        padding: 5,
        elevation: 12
    },
    songTextView: {
        padding: 8
    },
    songNameView: {
        fontSize: 20,
        color: 'black',
        fontWeight: 'bold'
    },
    songArtistsView: {
        fontSize: 16,
        color: 'black',
        fontWeight: 'bold'
    },
    songDurationView: {
        fontSize: 16,
        color: 'black'
    }

})

export default SongListTile;