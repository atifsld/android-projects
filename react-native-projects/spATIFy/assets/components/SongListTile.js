import react from "react";
import { View, Image, Text } from "react-native";
import { StyleSheet } from "react-native";

function SongListTile(song) {
    function logContents(song) {
        console.log("Tile receives:", song.song)
        console.log("song.songName:", song.song.songName)
    }
    return (
        <View style={styles.songCardView}>
            <View style={styles.songImageView}>
                <Image style = {styles.albumArt}
                    source={{uri: song.song.songAlbumArtUrl}}
                />
             </View>
             <View style={styles.songTextView}>
                <Text style={styles.songNameView}>{song.song.songName}</Text>
                <Text style={styles.songArtistsView}>{song.song.songArtists}</Text>
                <Text style={styles.songDurationView}>{song.song.songDuration}</Text>
             </View>
        </View>
    )
}

const styles = StyleSheet.create({
    songCardView: {
        flex: 1,
        flexDirection: 'row',
        margin: 12,
        height: 100,
        backgroundColor: "#03DAC6",
        elevation: 8,
        borderRadius: 8
    },
    albumArt: {
        height: 90,
        width: 90,
        borderRadius: 6
    },
    songImageView: {
        padding: 8,
        elevation: 12
    },
    songTextView: {
        padding: 8
    },
    songNameView: {
        fontSize: 20,
        fontWeight: 'bold'
    },
    songArtistsView: {
        fontSize: 16,
        fontWeight: 'bold'
    },
    songDurationView: {
        fontSize: 16
    }

})

export default SongListTile;