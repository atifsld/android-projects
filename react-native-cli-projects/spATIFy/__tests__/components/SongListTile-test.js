import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import SongListTile from '../../components/SongListTile'

const sampleSong = {
    "id": "d3d6ca38-cf1b-4151-9937-dd35cff2a336",
    "songAlbumId": "2080f161-c553-408d-9724-9c1e2f1b8c87",
    "songArtists": "Miles Davis",
    "songAlbumArtUrl": "https://ia802703.us.archive.org/26/items/mbid-e7ba3cb7-a074-45ee-870f-3baeb6d3e8bf/mbid-e7ba3cb7-a074-45ee-870f-3baeb6d3e8bf-12708426541.jpg",
    "songSpotifyUrl": "https://open.spotify.com/track/0TUa7gBg7LJH6B8dkNVioU?si=d23c31e6e9e24c5c",
    "songDuration": "11:32",
    "songName": "All Blues",
    "songTrackNumber": 4, 
    "songIsFavorite": false
}

test('test song list tile', () => {
    const tree = renderer.create(<SongListTile song={sampleSong} />).toJSON();
    expect(tree).toMatchSnapshot();
})