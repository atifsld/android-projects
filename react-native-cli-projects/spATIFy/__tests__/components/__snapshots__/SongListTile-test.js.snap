// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test song list tile 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#03DAC6",
        "borderRadius": 8,
        "elevation": 8,
        "flex": 1,
        "flexDirection": "row",
        "height": 100,
        "margin": 8,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 8,
      }
    }
  >
    <View
      style={
        Object {
          "elevation": 12,
          "padding": 5,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://ia802703.us.archive.org/26/items/mbid-e7ba3cb7-a074-45ee-870f-3baeb6d3e8bf/mbid-e7ba3cb7-a074-45ee-870f-3baeb6d3e8bf-12708426541.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 6,
            "height": 90,
            "width": 90,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "padding": 8,
          "width": 200,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "ellipsizeMode": "tail",
            "fontSize": 19,
            "fontWeight": "bold",
            "width": "100%",
          }
        }
      >
        All Blues
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "ellipsizeMode": "tail",
            "fontSize": 16,
            "fontWeight": "bold",
            "numberOfLines": 1,
            "width": "80%",
          }
        }
      >
        Miles Davis
      </Text>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "ellipsizeMode": "tail",
            "fontSize": 16,
            "numberOfLines": 1,
            "width": "80%",
          }
        }
      >
        11:32
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "paddingHorizontal": 20,
        }
      }
    />
  </View>
</View>
`;
